crypto/asn1/x_long.c

```diff
@@ -76,7 +76,7 @@ static int long_i2c(ASN1_VALUE **pval, unsigned char *cont, int *putype,
      * set.
      */
     if (ltmp < 0)
-        utmp = -(unsigned long)ltmp - 1;
+        utmp = 0 - (unsigned long)ltmp - 1;
```

Comment:
Hmm... As compiler will be forced to promote constants to unsigned long one can wonder if it's free to choose between a) `OUL - (unsigned long)ltmp - 1UL` and b) `(unsigned long)(0 - 1) - (unsigned long)ltmp`, or does it have to one of them? Not that it matters [on two's complement]...